### PR TITLE
chore: ci for example notebooks

### PR DESCRIPTION
**Pull request by anthonycorletti**
evolution of https://github.com/rungalileo/examples/pull/2

changes include: 
- a function in each notebook `_minimize_for_ci` that enables env var control of whether or not to run a notebook with downsampling
- the downsampling logic for each notebook

hf ner integration will break without the alpha build (0.7.1a5) so in ci we currently `pip install dataquality==0.7.1a5` which should be removed soon, once changes make it into 0.7.1

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,379 @@
+name: test
+
+on:
+  push:
+    branches:
+      # - main
+      - chore/ci-for-example-notebooks
```
Comment (bogdan-galileo) [inline .github/workflows/test.yaml:7]: Do you have to change this to main for the tests to get triggered ?

Comment (anthonycorletti) [reply .github/workflows/test.yaml:7]: yup!

Comment (anthonycorletti) [reply .github/workflows/test.yaml:7]: will change it back before merging

File: examples/named_entity_recognition/Named_Entity_Recognition_with_Huggingface_Trainer_and_🔭_Galileo.ipynb [hunk 2]
```diff
@@ -22,6 +22,13 @@
       "outputs": [],
```
Comment (bogdan-galileo) [inline examples/named_entity_recognition/Named_Entity_Recognition_with_Huggingface_Trainer_and_🔭_Galileo.ipynb:22]: <pre class="ql-syntax" spellcheck="false">Line #41.        f"task-{task}",
</pre><p>Nice catch!!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rungalileo/examples/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

File: examples/named_entity_recognition/Named_Entity_Recognition_with_Pytorch_and_🔭_Galileo.ipynb [hunk 3]
```diff
@@ -23,6 +23,13 @@
       "outputs": [],
```
Comment (bogdan-galileo) [inline examples/named_entity_recognition/Named_Entity_Recognition_with_Pytorch_and_🔭_Galileo.ipynb:23]: <p>I think we should look into trying to hide that better and not have it show up in the training loop. I'm worried that will confuse data scientists, especially that most of us see CI testing like that for the first time! Wdyt ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rungalileo/examples/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

File: examples/named_entity_recognition/Named_Entity_Recognition_with_Tensorflow_and_🔭_Galileo.ipynb [hunk 4]
```diff
@@ -23,6 +23,14 @@
       "outputs": [],
```
Comment (bogdan-galileo) [inline examples/named_entity_recognition/Named_Entity_Recognition_with_Tensorflow_and_🔭_Galileo.ipynb:23]: <p>Same, we try to hide that ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rungalileo/examples/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

**Comment by anthonycorletti** [comment]
will wait until dq 0.7.3 is released to merge

File: examples/named_entity_recognition/Named_Entity_Recognition_with_Tensorflow_and_🔭_Galileo.ipynb (hunk 4 above)
Comment (bogdan-galileo) [inline examples/named_entity_recognition/Named_Entity_Recognition_with_Tensorflow_and_🔭_Galileo.ipynb:23]: <p>Seems like we don't even use <code>data["validation"]</code> actually in this one. We create <code>tf_validation_set</code> but dont do anything with it! We can just delete it and only keep <code>train</code> and <code>test</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rungalileo/examples/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

**Review by bogdan-galileo — COMMENTED** [review]
We can try to hide the _minimize_for_ci commands in the hidden cells loading the data at some point if they look too verbose and confuse user.

**Review by bogdan-galileo — APPROVED** [review]
Super awesome to have these tests! Thankssss Anthony 🙏

**Pull request merged**